File: config/config-sbc2ha/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/platform/bus/BusConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.platform.bus;

import com.dfi.sbc2ha.config.sbc2ha.definition.platform.PlatformConfig;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public abstract class BusConfig extends PlatformConfig {
    @JsonProperty("bus_id")
    String busId;

}

File: config/editor/src/app/definition/platform/bus/BusConfig.ts
import {PlatformConfig} from "../PlatformConfig";
import {PlatformType} from "../../enums/PlatformType";
import {Expose} from "class-transformer";

export abstract class BusConfig extends PlatformConfig {
    @Expose({name: "bus_id"})
    busId: string;

    constructor(platform: PlatformType, busId: string) {
        super(platform);
        this.busId = busId;
    }
}
